File: app/src/main/java/com/donald/musictheoryapp/Music/ScoreView/SNoteHead.java
package com.donald.musictheoryapp.Music.ScoreView;

import android.graphics.Canvas;

import com.donald.musictheoryapp.Music.MusicXML.Note;
import com.donald.musictheoryapp.Music.MusicXML.Clef;

class SNoteHead extends ScoreStamp
{
	private char m_Glyph;
	private float verticalOffsetOnStaff;
	private float horizontalOffsetOnStaff;

	SNoteHead(ScoreView view)
	{
		super(view);
		m_Glyph = U.NOTE_HEAD_WHOLE;
		verticalOffsetOnStaff = 0;
	}

	protected void setNote(Note note, Clef clef)
	{
		//int notePosition = note.pitch.step + note.pitch.octave * Note.Pitch.Step.NO_OF_STEPS;
		setType(note.type);
		int noteStaffPosition = Clef.noteStaffPosition(note, clef);
		verticalOffsetOnStaff = -noteStaffPosition * staffStepHeight() - staffHeight();
		//m_VerticalOffsetOnStaff =
		//	(notePosition - Clef.Sign.BASE_NOTE_POS_BY_SIGN[clef.sign] + (clef.line - 1) * 2) * staffStepHeight();
	}

	private void setType(int type)
	{
		switch(type)
		{
			case Note.Type.BREVE:
				m_Glyph = U.NOTE_HEAD_D_WHOLE;
				horizontalOffsetOnStaff = -noteWidth() * 1.4f / 2;
				break;
			case Note.Type.WHOLE:
				m_Glyph = U.NOTE_HEAD_WHOLE;
				horizontalOffsetOnStaff = -noteWidth() * 1.4f / 2;
				break;
			case Note.Type.HALF:
				m_Glyph = U.NOTE_HEAD_HALF;
				horizontalOffsetOnStaff = -noteWidth() / 2;
				break;
			case Note.Type.QUARTER:
			case Note.Type.EIGHTH:
			case Note.Type.SIXTEENTH:
			case Note.Type.THIRTY_SECOND:
			case Note.Type.SIXTY_FOURTH:
				m_Glyph = U.NOTE_HEAD_BLACK;
				horizontalOffsetOnStaff = -noteWidth() / 2;
				break;
			default:
				throw new IllegalArgumentException(type + " is not a valid type.");
		}
	}

	@Override
	protected void onDraw(Canvas canvas, float posX, float posY)
	{
		canvas.drawText(String.valueOf(m_Glyph),
			posX + horizontalOffsetOnStaff,
			posY + glyphSize() + verticalOffsetOnStaff,
			glyphPaint()
		);
	}
}

File: app/src/main/java/com/donald/musictheoryapp/music/scoreview/SNoteHead.kt
package com.donald.musictheoryapp.music.scoreview

import android.graphics.Canvas
import com.donald.musictheoryapp.music.MusicXML.Clef
import com.donald.musictheoryapp.music.MusicXML.Clef.Companion.noteStaffPosition
import com.donald.musictheoryapp.music.MusicXML.Note
import java.lang.IllegalArgumentException

internal class SNoteHead(view: ScoreView) : ScoreStamp(view) {

    private var glyph = U.NOTE_HEAD_WHOLE
    var note: Note? = null
    var clef: Clef? = null
    private var noteStaffPosition = 0
    override var relContentWidth = 0F
        private set

    fun set(note: Note, clef: Clef) {
        when (note.type) {
            Note.Type.BREVE -> {
                glyph = U.NOTE_HEAD_D_WHOLE
                relContentWidth = REL_HOLLOW_NOTE_WIDTH
            }
            Note.Type.WHOLE -> {
                glyph = U.NOTE_HEAD_WHOLE
                relContentWidth = REL_HOLLOW_NOTE_WIDTH
            }
            Note.Type.HALF -> {
                glyph = U.NOTE_HEAD_HALF
                relContentWidth = REL_SOLID_NOTE_WIDTH
            }
            Note.Type.QUARTER, Note.Type.EIGHTH, Note.Type.SIXTEENTH, Note.Type.THIRTY_SECOND, Note.Type.SIXTY_FOURTH -> {
                glyph = U.NOTE_HEAD_BLACK
                relContentWidth = REL_SOLID_NOTE_WIDTH
            }
            else -> {
                glyph = U.NOTE_HEAD_WHOLE
                noteStaffPosition = 0
                relContentWidth = 0F
                throw IllegalArgumentException("${note.type} is not a valid type.")
            }
        }
        noteStaffPosition = noteStaffPosition(note, clef)
    }

    override fun onDraw(canvas: Canvas, posX: Float, posY: Float) {
        //val note = note ?: return
        //val clef = clef ?: return
        //set(note, clef)
        val verticalOffsetOnStaff = -noteStaffPosition * staffStepHeight + staffHeight
        canvas.drawText(
            glyph.toString(),
            posX,
            posY + verticalOffsetOnStaff,
            glyphPaint
        )
    }

}